From: Liam Howlett <liam.howlett@oracle.com>
Subject: maple_tree: drop typedef from header
Date: Tue, 19 Jul 2022 17:10:15 +0000

These typedefs are not used and cause issues with rust in the kernel.

Link: https://lkml.kernel.org/r/20220719171006.3299123-1-Liam.Howlett@oracle.com
Fixes: fb7297e8a66b ("Maple Tree: add new data structure")
Signed-off-by: Liam R. Howlett <Liam.Howlett@oracle.com>
Reported-by: Li Hongyu <lihongyu1999@bupt.edu.cn>
Suggested-by: Matthew Wilcox <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/maple_tree.h |    2 --
 1 file changed, 2 deletions(-)

--- a/include/linux/maple_tree.h~maple-tree-add-new-data-structure-fix-2
+++ a/include/linux/maple_tree.h
@@ -72,8 +72,6 @@
  *   0x010 : 32 bit values, type in 0-2, slot in 3-6
  *   0x110 : 64 bit values, type in 0-2, slot in 3-6
  */
-typedef struct maple_enode *maple_enode; /* encoded node */
-typedef struct maple_pnode *maple_pnode; /* parent node */
 
 /*
  * This metadata is used to optimize the gap updating code and in reverse
_
